src/lexer.mll

```diff
@@ -675,11 +673,29 @@ and code_block_metadata_tail input = parse
   | eof
     { `Eof }
 
-and code_block start_offset metadata input = parse
-  | (code_block_text as c) "]}"
-    { emit_code_block ~start_offset input metadata c }
-  | (code_block_text as c) eof
+and code_block start_offset content_offset metadata prefix delim input = parse
+  | ("]" (language_tag_char* as delim') "[") as terminator
```

Comment:
We should have a `delim_char`. Perhaps without `-` because `{-` exsists.

Reply:
Good plan